hostinfo/hostinfo_linux.go

```diff
@@ -48,6 +49,15 @@ func linuxDeviceModel() string {
 	return ""
 }
 
+func getQnapQtsVersion(version_info []byte) string {
+	re := regexp.MustCompile(`QTSFW_([\d.]+)`)
```

Comment:
We shouldn't compile on each call. Compiling the regexp is quite heavy.

OTOH, we shouldn't compile it in init either, ideally, especially as most systems aren't QNAP.

I see that GetOSVersion isn't cached (except on some platforms that call SetOSVersion) so that means this will be run on every call to hostinfo.New, which I think is often enough. (new map calls, etc)

Maybe we should add some caching to osVersion Linux at least. It's not super cheap currently in any case.

Not your cleanup, though. You can just add a TODO here, like `// TODO(bradfitz,dgentry): cache this or caller(s)`

Reply:
Made the regex compile once the first time it is used, added a TODO about caching at a higher layer.